internal/datastore/memdb/readwrite.go

```diff
@@ -87,6 +96,37 @@ func (rwt *memdbReadWriteTx) write(tx *memdb.Txn, mutations ...*core.RelationTup
 	return nil
 }
 
+func (rwt *memdbReadWriteTx) prepareCaveat(tx *memdb.Txn, mutation *core.RelationTupleUpdate) (*caveatReference, error) {
+	var cr *caveatReference
+	if mutation.Tuple.Caveat != nil {
```

Comment:
So, my plan for this:

1. Every caveat written into the system gets a unique `uint64` ID
2. When the schema is compiled, a name -> ID map is written alongside the type information
3. When we perform type checking in the `WriteRelationships` call, we'll need to perform a lookup by name against the schema, which will give us back the ID
4. The caveat *ID* should be placed into the `RelationTupleUpdate` and then we can just use it directly here without further queries

Reply:
I've added caveat ID and lookup methods in https://github.com/authzed/spicedb/pull/807/commits/d2d7ce075cb08cd260011d79057aed85674df9f9